statsmodels/discrete/discrete_model.py

```diff
@@ -2371,11 +2371,10 @@ def __init__(self, model, mlefit, cov_type='nonrobust', cov_kwds=None,
 
 
     def __getstate__(self):
-        try:
-            #remove unpicklable callback
-            self.mle_settings['callback'] = None
-        except (AttributeError, KeyError):
-            pass
+        # remove unpicklable methods
+        self.mle_settings['callback'] = None
```

Comment:
Silly me, that's in `discrete_model` and not in `base` All discrete models use currently scipy optimizers and have mle_setting, so there is no failing test case here.